Gateway/Request/BasicParameter/SaveOrderBeforeDataBuilder.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Buckaroo\Magento2\Gateway\Request\BasicParameter;
+
+use Buckaroo\Magento2\Gateway\Helper\SubjectReader;
+use Buckaroo\Magento2\Model\ConfigProvider\Account;
+use Magento\Payment\Gateway\Request\BuilderInterface;
+use Magento\Sales\Model\Order;
+
+class SaveOrderBeforeDataBuilder implements BuilderInterface
+{
+    /**
+     * @var Account
+     */
+    protected $configProviderAccount;
+
+    /**
+     * @param Account $configProviderAccount
+     */
+    public function __construct(Account $configProviderAccount)
+    {
+        $this->configProviderAccount = $configProviderAccount;
+    }
+
+    /**
+     * Save Order Before Request
+     *
+     * @param array $buildSubject
+     * @return array|void
+     */
+    public function build(array $buildSubject)
+    {
+        $paymentDO = SubjectReader::readPayment($buildSubject);
+        /** @var Order $order */
+        $order = $paymentDO->getOrder()->getOrder();
+        $store = $order->getStoreId();
+
+        if ($this->configProviderAccount->getCreateOrderBeforeTransaction($store)) {
+            $newStatus = $this->configProviderAccount->getOrderStatusNew($store);
+            $orderState = 'new';
+            if (!$newStatus) {
+                $newStatus = $order->getConfig()->getStateDefaultStatus($orderState);
+            }
+
+            $order->setState($orderState);
```

Comment:
why do we need to set the status/state here?

Reply:
The logic was copied from here: app/code/Buckaroo/Magento2/Gateway/Http/TransactionBuilder/Order.php:66
And orderStatus is set from configuration. For orderState I don't have explanation why is set on "New" if any order start with the state New.